js/src/modals/Shapeshift/shapeshift.js

```diff
@@ -0,0 +1,320 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { FlatButton } from 'material-ui';
+import ActionDoneAll from 'material-ui/svg-icons/action/done-all';
+import ContentClear from 'material-ui/svg-icons/content/clear';
+
+import { IdentityIcon, Modal } from '../../ui';
+import { newError } from '../../redux/actions';
+import initShapeshift from '../../3rdparty/shapeshift';
+import shapeshiftLogo from '../../images/shapeshift-logo.png';
+
+import AwaitingDepositStep from './AwaitingDepositStep';
+import AwaitingExchangeStep from './AwaitingExchangeStep';
+import CompletedStep from './CompletedStep';
+import ErrorStep from './ErrorStep';
+import OptionsStep from './OptionsStep';
+
+import styles from './shapeshift.css';
+
+const shapeshift = initShapeshift();
+
+const STAGE_NAMES = ['details', 'awaiting deposit', 'awaiting exchange', 'completed'];
+
+class FundAccount extends Component {
+  static propTypes = {
+    address: PropTypes.string.isRequired,
+    newError: PropTypes.func.isRequired,
+    onClose: PropTypes.func
+  }
+
+  state = {
+    stage: 0,
```

Comment:
I think strings as values of `stage` would be both more robust & easier to reason about.